src/app/data-explorer/data-specification.service.ts

```diff
@@ -542,34 +541,44 @@ export class DataSpecificationService {
   ): Observable<DataSpecification> {
     return forkJoin([
       // TODO: assume there is only one data model, will have to change in future
+      // Get the rootModel and create a new empty data specification.
       this.dataExplorer.getRootDataModel(),
       this.create(user, name, description),
     ]).pipe(
       switchMap(([rootDataModel, dataSpecification]) => {
-        if (!rootDataModel || !rootDataModel.id) {
-          return throwError(() => new Error('No root data model'));
+        if (!rootDataModel?.id || !dataSpecification?.id) {
+          return throwError(() => new Error('Missing root data model or data specification'));
```

Comment:
I don't like this. This makes it harder to work out what has gone wrong. Is it broken because the root model is missing or because the data specification is missing? I would prefer this check as separate errors: e.g:
```
        if (!rootDataModel?.id) {
          return throwError(() => new Error('Missing root data model'));

        if (!dataSpecification?.id) {
          return throwError(() => new Error('Missing data specification'));
```

Reply:
Reverted.